### ISSUE TITLE
Central place for plugins and scripts

### ISSUE THREAD

**Issue by APerson241** [hexchat/hexchat#2015]
It's not that easy to find a HexChat script for a given purpose right now. The hexchat/hexchat-addons repository has scripts, but they're not easily searchable. Some scripts and plugins are also scattered across other repositories, such as Poorchop/hexchat-scripts or TingPing/plugins.

The HexChat website and documentation do not tell users where to find scripts or plugins. We should make a website that has a list of as many scripts and plugins as possible, and that has descriptions of each.

We could use the GitHub wiki for hexchat or hexchat-addons, or the GitHub Pages feature for either repository. Any other suggestions?

**Comment by TingPing** [comment]
At a time we were going to make a website but that time has passed and all potential webdevs are gone. I don't for-see this happening any time soon.

**Comment by APerson241** [comment]
@TingPing Could we use the GitHub Pages domain of either this repository or the addons one? I'd be happy to make and update a pure-HTML website.

**Comment by TingPing** [comment]
I don't think so, the addons repo is mostly a dumping ground of unmaintained scripts and to pull in scripts from around the net requires a lot more maintenance, review, work, etc.

**Comment by APerson241** [comment]
Agreed; I'm volunteering to do that maintenance work, since I believe the existence of a central page, listing scripts from everywhere, not just the addons repo, would be a real benefit to users.

**Comment by TingPing** [comment]
There are a few questions that need to be answered:

How much data do we want here?

- Name and description of script
- License
- Language
- Author

What features does the site need?

We at least want it to be filterable and searchable.

What will the process be for adding new scripts to it?

**Comment by amezarashi97** [comment]
I'd be happy to help if needed.

**Comment by APerson241** [comment]
I was thinking just name and description (and a download link, of course), since the purpose is to serve HexChat users.

One issue here is we somehow need to find free, permanent web hosting that allows dynamic content (i.e. searching). GitHub Pages does not allow dynamic content. The GitHub wiki allows searching, so that's one option. Read the Docs is another great option that allows searching.

Ideally, we'd have a very low barrier to adding new scripts, to encourage development. A publicly-editable GitHub wiki works for this, and Read the Docs allows this as well (with pull requests).

**Comment by TingPing** [comment]
You can do client side searching with javascript.

**Comment by learn-more** [comment]
https://datatables.net/examples/api/multi_filter.html

**Comment by amezarashi97** [comment]
test

**Comment by amezarashi97** [comment]
Would you rather create a new page inside the existing layout that is hexchat.github.io or shall we create a new one?

**Comment by APerson241** [comment]
I was thinking of using hexchat.github.io/hexchat-addons, which will be created from the gh-pages branch of hexchat-addons. That way, we don't have to clutter up the commit history of the hexchat/hexchat.github.com repo.

**Comment by TingPing** [comment]
@APerson241 Sounds fine, we can rename the repo to `addons` for a cleaner URL too.

**Comment by APerson241** [comment]
I like the rename idea!

**Comment by TingPing** [comment]
Actually maybe even better is just a new `addons` repo. The `hexchat-addons` repo gives commit access to quite a few people.

**Comment by bil-elmoussaoui** [comment]
@TingPing A github page can be easier to handle. People can submit PR with new .md files that contain new plugins. Or even add that to the main web site of Hexchat

**Comment by learn-more** [comment]
Maybe someone can write a nice plugin manager that queries / searches this site and/or suggests plugins to install.

**Comment by TingPing** [comment]
@learn-more I have this but its rather janky: https://github.com/TingPing/plugins/blob/master/HexChat/script.py

**Comment by learn-more** [comment]
@TingPing : Is there something blocking this?

**Comment by TingPing** [comment]
Not really, anybody is free to work on this. As mentioned the main requirements is to just be a static site.
